core/tasty.cabal

```diff
@@ -64,6 +64,10 @@ library
     -- for GHC.Generics
     build-depends: ghc-prim
 
+  if impl(ghc < 8.0)
+    -- provide/emulate `Data.Semigroups` API for pre-GHC-8
+    build-depends: semigroups == 0.18.*
+
```

Comment:
They way I did it (with CPP pragmas) was intentional — I'd rather not add extra dependencies without a good reason.

Reply:
I guess I could avoid the new dependency defining `(<>) = mappend` but I prefer the current approach, that is, the GHC 8.4.1-alpha approach, because it follows the mathematical definitions.